src/main/python/hydra/lib/hydrabase.py

```diff
@@ -666,6 +671,66 @@ def get_mesos_slave_ips_attr(self, attr_type, attr_value):
         """
         return self.__mesos.get_slave_ips_from_attribute(attr_type, attr_value)
 
+    def get_all_mesos_slave_attr(self):
+        """
+        Get all slave attributes.
+        """
+        attr_list = []
+        for idx, info in self.mesos_cluster.items():
+            attr_list.append([info["cat"], info["match"]])
+        # A sample attribute list would be
+        # [['slave_id', 'slave-set1_0'], ['slave_id', 'slave-set1_1'], ['slave_id', 'slave-set1_2']]
+        return attr_list
+
+    def get_all_mesos_slave_iplist(self):
+        """
+        Get all mesos slave ips
+        """
+        ip_list = list()
+        attr_list = self.get_all_mesos_slave_attr()
+        for attr in attr_list:
+            slave_ips = self.get_mesos_slave_ips_attr(attr[0], attr[1])
+            if len(slave_ips) == 0:
+                raise Exception("NO slave exists with attribtue [%s=%s]" % (attr[0], attr[1]))
+            ip_list.append(slave_ips[0])
+        return ip_list
+
+    def get_app_mem_cpu_stats(self, name):
+        """
+        Get an apps stats querying all slaves
+        If multiple tasks are running for the same app
+        e-g an app x which is scaled to y instances.
+        This function will return a list of all tasks
+        stats for that app
+        @args:
+        name  :  Name of the app
+        """
+        app_stats_list = {}
+        task_stats = {"timestamp": 0,
+                      "mem_rss_bytes": 0,
+                      "cpu_system_time_secs": 0,
+                      "cpu_user_time_secs": 0,
+                      }
+        assert(name in self.app_id_list)
+        if not self.all_mesos_slave_iplist:
+            self.all_mesos_slave_iplist = self.get_all_mesos_slave_iplist()
```

Comment:
Yes, thats a valid exception. What I was trying to say was that when get_all_mesos_slave_iplist() will raise that exception, our test case will stop immediately. If we surround the function call in try catch statement, we can handle the exception in whatever way we want in Except part.